lib/typing/ext/parse.go

```diff
@@ -24,78 +24,48 @@ func ParseDateFromInterface(val any) (time.Time, error) {
 	switch convertedVal := val.(type) {
 	case time.Time:
 		return convertedVal, nil
-	case *ExtendedTime:
-		return convertedVal.GetTime(), nil
 	case string:
 		return parseDate(convertedVal)
 	default:
 		return time.Time{}, fmt.Errorf("unsupported type: %T", convertedVal)
 	}
 }
 
-func ParseTimestampNTZFromInterface(val any) (time.Time, error) {
+func ParseTimeFromInterface(val any) (time.Time, error) {
 	switch convertedVal := val.(type) {
+	case nil:
+		return time.Time{}, fmt.Errorf("val is nil")
 	case time.Time:
 		return convertedVal, nil
-	case *ExtendedTime:
-		return convertedVal.GetTime(), nil
 	case string:
-		return parseTimestampNTZ(convertedVal)
+		return parseTime(convertedVal)
 	default:
 		return time.Time{}, fmt.Errorf("unsupported type: %T", convertedVal)
 	}
 }
 
-func ParseTimestampTZFromInterface(val any) (time.Time, error) {
+func ParseTimestampNTZFromInterface(val any) (time.Time, error) {
 	switch convertedVal := val.(type) {
-	case nil:
-		return time.Time{}, fmt.Errorf("val is nil")
```

Comment:
Why isn't this one handling the nil case anymore?

Reply:
Not sure, can add back. Handling or not handling yield will both error